conans/test/functional/graph/test_graph_build_mode.py

```diff
@@ -0,0 +1,167 @@
+import pytest
+import operator
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.utils.tools import TestClient
+
+
+class fakeop:
+    def __init__(self, function):
+        self.function = function
+    def __ror__(self, other):
+        return fakeop(lambda x, self=self, other=other: self.function(other, x))
+    def __or__(self, other):
+        return self.function(other)
+    def __rlshift__(self, other):
+        return fakeop(lambda x, self=self, other=other: self.function(other, x))
+    def __rshift__(self, other):
+        return self.function(other)
+    def __call__(self, value1, value2):
+        return self.function(value1, value2)
+
+
+@pytest.fixture(scope="module")
+def build_all():
+    """ Build a simple graph to test --build option
+
+        foobar <- bar <- foo
+               <--------|
+
+        All packages are built from sources to keep a cache.
+    :return: TestClient instance
+    """
+    client = TestClient()
+    client.save({"conanfile.py": GenConanfile().with_setting("build_type")})
+    client.run("export . foo/1.0@user/testing")
+    client.save({"conanfile.py": GenConanfile().with_require("foo/1.0@user/testing")
+                .with_setting("build_type")})
+    client.run("export . bar/1.0@user/testing")
+    client.save({"conanfile.py": GenConanfile().with_require("foo/1.0@user/testing")
+                .with_require("bar/1.0@user/testing")
+                .with_setting("build_type")})
+    client.run("export . foobar/1.0@user/testing")
+    client.run("install foobar/1.0@user/testing --build")
+
+    return client
+
+
+def test_install_build_single(build_all):
+    """ When only --build=<ref> is passed, only <ref> must be built
+    """
+    build_all.run("install foobar/1.0@user/testing --build=foo")
```

Comment:
This should be extensively tested by now, no need for a new test?

Reply:
It's possible yes, we already have some tests in graph build, but we can add some pattern here. My idea here is for regression, I don't want to break the old behavior, but we don't have many tests which check it.

Reply:
You are right, seems not that covered, better be on the safe side